dts/common/yaml/zephyr-i2c.yaml

```diff
@@ -0,0 +1,19 @@
+---
+title: Zephyr specific I2C properties
+id: i2c
+version: 0.1
+
+description: >
+    This binding gives Zephyr specific I2C properties
+
+properties:
+  - zephyr,speed :
+      type: int
+      category: required
+      description: Speed setting (1..5) as defined in i2c.h
+      generation: define
+  - zephyr,use-10-bit-addr :
```

Comment:
I understand your point, though,here, we're setting a default configuration at boot (as meant initial flag).
This doesn't prevent to change it in run time, as allowed by api.
It's a bit same as UART's current-speed... 
My advice would be to keep it, but make it clear in description this is  a default boot setting.